### ISSUE TITLE
Asking more questions which are not just related to implementation.

### ISSUE THREAD

**Issue by Hirmay** [qiskit-community/qiskit-textbook#655]
**Is your feature request related to a problem? Please describe.**
For example if you see [Grover's algorithm](https://qiskit.org/textbook/ch-algorithms/grover.html#problems) section in qiskit textbook, you would see in problems part there are two questions. Both about modifying it to 2 or 4 qubits seeing if the results are what you expect.

**Describe the solution you'd like**
Instead we could have more interesting and intriguing questions that would certainly help the readers understand the concept in much more detail and gave them good clarity.

**Describe alternatives you've considered**
As an example if we have question like, Consider the case where  **N/4**  elements are marked instead of just one. If we run **one** iteration of Grover's algorithm and measure, what is the probability that we **see a marked element**?

**Additional context**
Add any other context or screenshots about the request here.

**Comment by Hirmay** [comment]
If you think it's a good enhancement I would love to contribute towards it.

**Comment by aditya-giri** [comment]
I second this. It would be very useful for developing an intuitive understanding of the algorithm. There's also a good visualization tool for Grover's that we can make use of (which I believe was made during the summer school): http://megoapps.com/grover/.

**Comment by Hirmay** [comment]
Thank you for your support and this tool seems very good. I think first playing with this tool and solving interesting questions about Grover's algorithm will certainly help to develop great intuitive understanding of the algorithm.

**Comment by Hirmay** [comment]
I was thinking if you don't mind @aditya-giri we can collaborate on this. If you think @frankharkins if this is a good enhancement could you assign this to us so we can start working on this?

**Comment by frankharkins** [comment]
I agree the problems could be updated, but I think #665 needs addressing first. I otherwise worry that the problems might not follow naturally from the new content. Let me work on this early next week

**Comment by frankharkins** [comment]
@aditya-giri @Hirmay some of the questions have been updated as part of a review of the Grover chapter, I would appreciate feedback on this as I tried to take your comments into account when re-writing the problems. https://qiskit.org/textbook/ch-algorithms/grover.html

**Comment by Hirmay** [comment]
You have done a fantastic job where the questions are significantly better than the ones that were asked before. I have two more suggestions not just for Grover's algorithm but in general,
One is can we add more questions on the theoretical side like the one I mentioned which was "Consider the case where N/4 elements are marked instead of just one. If we run one iteration of Grover's algorithm and measure, what is the probability that we see a marked element?" as I don't see questions like that.
Second is can we add some MCQs or some other thing which will make it more interactive for students and they'll be more engaged in it.
If you think those are valid things we could implement, I would love to help you in implementing them.

**Comment by aditya-giri** [comment]
Love the new questions - the sudoku is very cool!

**Comment by frankharkins** [comment]
@Hirmay that sounds good! Do you want to add those and make a PR?

**Comment by Hirmay** [comment]
I'll have to PR the entire textbook right as I was thinking we can implement those two suggestions in many sections of the textbook or do you want me to just do a PR on Grover's part for now?

**Comment by aditya-giri** [comment]
I think it makes more sense to start with Grover's and take the others incrementally...

**Comment by frankharkins** [comment]
@Hirmay yes let's stick with Grover for now

**Comment by Hirmay** [comment]
Okay so I'll just prepare some questions(like 3-4) for Grover's algorithm and later we can incorporate them into interactive MCQs / quizzes form.

**Comment by Hirmay** [comment]
So sorry I forgot about this. At that time, I got covid and then had some pending academic stuff, so it slipped out of my mind.

I have provided some questions below, and please go through them. I would appreciate any feedback you have regarding this!

1. Does the phase inversion operation matrix is itself the inverse?
**Yes** 
No
Explanation: The phase inversion operator flips the phase of the marked element (M). Hence, if you apply the phase inversion twice, you return to the original superposition.

2. Let us consider a case where N/4 elements are marked. Given that there is only one iteration of Grover's algorithm, what is the probability that we see one of the marked elements?
1/4
1/3
1/2
**1**
Explanation: I have to write this in latex, which Github doesn't support.

3. Is the matrix produced by the operation of inverting around the mean itself the inverse?
**Yes** 
No
Explanation: Since we are reflecting every state concerning the mean, if we apply it twice, we return to the initial superposition.
 
4. What is the time complexity of Grover's algorithm if there are M-marked elements out of the N total elements?
O(sqrt(N))
O(sqrt(M/N))
O(N)
**O(sqrt(N/M))**

#NOTE: I can produce interactive MCQs, but only in flask. Qiskit probably accepts Javascript, but I'm not familiar with that.

**Comment by frankharkins** [comment]
Thanks for your contribution. I'm closing this issue as this repository is being archived. You can find the latest version of the Qiskit Textbook online at <https://qiskit.org/learn/>, and the new content repository at [Qiskit/textbook](https://github.com/Qiskit/textbook). Please feel free to re-open this issue there if it's still relevant.
